etna/transforms/base.py

```diff
@@ -1,20 +1,182 @@
+        return ts
+
+    @abstractmethod
+    def _fit(self, df: pd.DataFrame) -> "NewTransform":
+        """Fit the transform.
+
+        Should be implemented by user.
+
+        Parameters
+        ----------
+        df:
+            Dataframe in etna wide format.
+
+        Returns
+        -------
+        :
+            The fitted transform instance.
+        """
+        pass
+
+    def fit(self, ts: TSDataset) -> "NewTransform":
+        """Fit the transform.
+
+        Parameters
+        ----------
+        ts:
+            Dataset to fit the transform on.
+
+        Returns
+        -------
+        :
+            The fitted transform instance.
+        """
+        df = ts.to_pandas(flatten=False, features=self.required_features)
+        self._fit(df=df)
+        return self
+
+    @abstractmethod
+    def _transform(self, df: pd.DataFrame) -> pd.DataFrame:
+        """Transform dataframe.
+
+        Should be implemented by user
+
+        Parameters
+        ----------
+        df:
+            Dataframe in etna wide format.
+
+        Returns
+        -------
+        :
+            Transformed Dataframe in etna wide format.
+        """
+        pass
+
+    def transform(self, ts: TSDataset) -> TSDataset:
+        """Transform TSDataset inplace.
+
+        Parameters
+        ----------
+        ts:
+            Dataset to transform.
+
+        Returns
+        -------
+        :
+            Transformed TSDataset.
+        """
+        df = ts.to_pandas(flatten=False, features=self.required_features)
+        df_transformed = self._transform(df=df)
+        ts = self._update_dataset(ts=ts, df=df, df_transformed=df_transformed)
+        return ts
+
+    def fit_transform(self, ts: TSDataset) -> TSDataset:
+        """Fit and transform TSDataset.
+
+        May be reimplemented. But it is not recommended.
+
+        Parameters
+        ----------
+        ts:
+            TSDataset to transform.
+
+        Returns
+        -------
+        :
+            Transformed TSDataset.
+        """
+        return self.fit(ts=ts).transform(ts=ts)
+
+    def _inverse_transform(self, df: pd.DataFrame) -> pd.DataFrame:
+        """Inverse transform dataframe.
+
+        Parameters
+        ----------
+        df:
+            Dataframe in etna wide format.
 
-    in_column = "target"
+        Returns
+        -------
+        :
+            Dataframe in etna wide format after applying inverse transformation.
+        """
+        return df
+
+    def inverse_transform(self, ts: TSDataset) -> TSDataset:
+        """Inverse transform TSDataset.
+
+        Should be reimplemented in the classes with reimplemented _inverse_transform method.
+
+        Parameters
+        ----------
+        ts:
+            TSDataset to be inverse transformed.
+
+        Returns
+        -------
+        :
+            TSDataset after applying inverse transformation.
+        """
+        return ts
```

Comment:
Is it ok that it just returns `ts`?

Reply:
There are many transforms that do nothing in inverse_transform, I suggest to implement the inverse_transform in each transform where it is necessary.

Reply:
Add the common logic in the current implementation